### ISSUE TITLE
cusparse.h

### ISSUE THREAD

**Issue by patrafter1999** [vllm-project/vllm#755]
The package generates an error in Ubuntu in my Databricks instance where CUDA is installed. I think the problem is related to CUDA. Any idea?

```!pip install vllm```

```
Collecting vllm
  Downloading vllm-0.1.3.tar.gz (102 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 102.7/102.7 kB 2.3 MB/s eta 0:00:00a 0:00:01
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting ninja
  Using cached ninja-1.11.1-py2.py3-none-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (145 kB)
Collecting uvicorn
  Downloading uvicorn-0.23.2-py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 kB 388.2 kB/s eta 0:00:00 0:00:01
Collecting torch>=2.0.0
  Using cached torch-2.0.1-cp310-cp310-manylinux1_x86_64.whl (619.9 MB)
Requirement already satisfied: pydantic<2 in /databricks/python3/lib/python3.10/site-packages (from vllm) (1.10.6)
Requirement already satisfied: transformers>=4.31.0 in /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages (from vllm) (4.32.0.dev0)
Requirement already satisfied: psutil in /databricks/python3/lib/python3.10/site-packages (from vllm) (5.9.0)
Collecting fastapi
  Downloading fastapi-0.101.0-py3-none-any.whl (65 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.7/65.7 kB 430.9 kB/s eta 0:00:00a 0:00:01
Collecting ray>=2.5.1
  Downloading ray-2.6.2-cp310-cp310-manylinux2014_x86_64.whl (56.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.9/56.9 MB 9.2 MB/s eta 0:00:00:00:0100:01
Requirement already satisfied: numpy in /databricks/python3/lib/python3.10/site-packages (from vllm) (1.21.5)
Collecting xformers>=0.0.19
  Downloading xformers-0.0.20-cp310-cp310-manylinux2014_x86_64.whl (109.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 109.1/109.1 MB 5.5 MB/s eta 0:00:0000:0100:01
Requirement already satisfied: sentencepiece in /databricks/python3/lib/python3.10/site-packages (from vllm) (0.1.97)
Requirement already satisfied: typing-extensions>=4.2.0 in /databricks/python3/lib/python3.10/site-packages (from pydantic<2->vllm) (4.3.0)
Requirement already satisfied: pyyaml in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (6.0)
Requirement already satisfied: requests in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (2.28.1)
Requirement already satisfied: msgpack<2.0.0,>=1.0.0 in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (1.0.5)
Requirement already satisfied: grpcio>=1.42.0 in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (1.48.1)
Requirement already satisfied: packaging in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (21.3)
Requirement already satisfied: aiosignal in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (1.3.1)
Requirement already satisfied: jsonschema in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (4.16.0)
Requirement already satisfied: frozenlist in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (1.3.3)
Requirement already satisfied: click>=7.0 in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (8.0.4)
Requirement already satisfied: filelock in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (3.6.0)
Requirement already satisfied: protobuf!=3.19.5,>=3.15.3 in /databricks/python3/lib/python3.10/site-packages (from ray>=2.5.1->vllm) (3.19.4)
Collecting nvidia-cusolver-cu11==11.4.0.1
  Using cached nvidia_cusolver_cu11-11.4.0.1-2-py3-none-manylinux1_x86_64.whl (102.6 MB)
Requirement already satisfied: networkx in /databricks/python3/lib/python3.10/site-packages (from torch>=2.0.0->vllm) (2.8.4)
Collecting nvidia-cuda-cupti-cu11==11.7.101
  Using cached nvidia_cuda_cupti_cu11-11.7.101-py3-none-manylinux1_x86_64.whl (11.8 MB)
Requirement already satisfied: jinja2 in /databricks/python3/lib/python3.10/site-packages (from torch>=2.0.0->vllm) (2.11.3)
Collecting triton==2.0.0
  Using cached triton-2.0.0-1-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (63.3 MB)
Collecting nvidia-curand-cu11==10.2.10.91
  Using cached nvidia_curand_cu11-10.2.10.91-py3-none-manylinux1_x86_64.whl (54.6 MB)
Collecting nvidia-cudnn-cu11==8.5.0.96
  Using cached nvidia_cudnn_cu11-8.5.0.96-2-py3-none-manylinux1_x86_64.whl (557.1 MB)
Collecting nvidia-cuda-runtime-cu11==11.7.99
  Using cached nvidia_cuda_runtime_cu11-11.7.99-py3-none-manylinux1_x86_64.whl (849 kB)
Collecting nvidia-cublas-cu11==11.10.3.66
  Using cached nvidia_cublas_cu11-11.10.3.66-py3-none-manylinux1_x86_64.whl (317.1 MB)
Collecting nvidia-cusparse-cu11==11.7.4.91
  Using cached nvidia_cusparse_cu11-11.7.4.91-py3-none-manylinux1_x86_64.whl (173.2 MB)
Collecting nvidia-cufft-cu11==10.9.0.58
  Using cached nvidia_cufft_cu11-10.9.0.58-py3-none-manylinux1_x86_64.whl (168.4 MB)
Collecting nvidia-cuda-nvrtc-cu11==11.7.99
  Using cached nvidia_cuda_nvrtc_cu11-11.7.99-2-py3-none-manylinux1_x86_64.whl (21.0 MB)
Collecting nvidia-nvtx-cu11==11.7.91
  Using cached nvidia_nvtx_cu11-11.7.91-py3-none-manylinux1_x86_64.whl (98 kB)
Collecting sympy
  Using cached sympy-1.12-py3-none-any.whl (5.7 MB)
Collecting nvidia-nccl-cu11==2.14.3
  Using cached nvidia_nccl_cu11-2.14.3-py3-none-manylinux1_x86_64.whl (177.1 MB)
Requirement already satisfied: setuptools in /databricks/python3/lib/python3.10/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch>=2.0.0->vllm) (63.4.1)
Requirement already satisfied: wheel in /databricks/python3/lib/python3.10/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch>=2.0.0->vllm) (0.37.1)
Collecting cmake
  Using cached cmake-3.27.2-py2.py3-none-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (26.1 MB)
Collecting lit
  Using cached lit-16.0.6-py3-none-any.whl
Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in /databricks/python3/lib/python3.10/site-packages (from transformers>=4.31.0->vllm) (0.13.3)
Requirement already satisfied: huggingface-hub<1.0,>=0.15.1 in /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages (from transformers>=4.31.0->vllm) (0.16.4)
Requirement already satisfied: tqdm>=4.27 in /databricks/python3/lib/python3.10/site-packages (from transformers>=4.31.0->vllm) (4.64.1)
Requirement already satisfied: safetensors>=0.3.1 in /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages (from transformers>=4.31.0->vllm) (0.3.2)
Requirement already satisfied: regex!=2019.12.17 in /databricks/python3/lib/python3.10/site-packages (from transformers>=4.31.0->vllm) (2022.7.9)
Collecting pyre-extensions==0.0.29
  Downloading pyre_extensions-0.0.29-py3-none-any.whl (12 kB)
Requirement already satisfied: typing-inspect in /databricks/python3/lib/python3.10/site-packages (from pyre-extensions==0.0.29->xformers>=0.0.19->vllm) (0.8.0)
Collecting starlette<0.28.0,>=0.27.0
  Downloading starlette-0.27.0-py3-none-any.whl (66 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 67.0/67.0 kB 643.0 kB/s eta 0:00:00a 0:00:01
Collecting typing-extensions>=4.2.0
  Using cached typing_extensions-4.7.1-py3-none-any.whl (33 kB)
Collecting h11>=0.8
  Downloading h11-0.14.0-py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.3/58.3 kB 12.2 MB/s eta 0:00:00
Requirement already satisfied: six>=1.5.2 in /usr/lib/python3/dist-packages (from grpcio>=1.42.0->ray>=2.5.1->vllm) (1.16.0)
Requirement already satisfied: fsspec in /databricks/python3/lib/python3.10/site-packages (from huggingface-hub<1.0,>=0.15.1->transformers>=4.31.0->vllm) (2022.7.1)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /databricks/python3/lib/python3.10/site-packages (from packaging->ray>=2.5.1->vllm) (3.0.9)
Collecting anyio<5,>=3.4.0
  Downloading anyio-3.7.1-py3-none-any.whl (80 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.9/80.9 kB 17.0 MB/s eta 0:00:00
Requirement already satisfied: MarkupSafe>=0.23 in /databricks/python3/lib/python3.10/site-packages (from jinja2->torch>=2.0.0->vllm) (2.0.1)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /databricks/python3/lib/python3.10/site-packages (from jsonschema->ray>=2.5.1->vllm) (0.18.0)
Requirement already satisfied: attrs>=17.4.0 in /databricks/python3/lib/python3.10/site-packages (from jsonschema->ray>=2.5.1->vllm) (21.4.0)
Requirement already satisfied: certifi>=2017.4.17 in /databricks/python3/lib/python3.10/site-packages (from requests->ray>=2.5.1->vllm) (2022.9.14)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /databricks/python3/lib/python3.10/site-packages (from requests->ray>=2.5.1->vllm) (1.26.11)
Requirement already satisfied: idna<4,>=2.5 in /databricks/python3/lib/python3.10/site-packages (from requests->ray>=2.5.1->vllm) (3.3)
Requirement already satisfied: charset-normalizer<3,>=2 in /databricks/python3/lib/python3.10/site-packages (from requests->ray>=2.5.1->vllm) (2.0.4)
Collecting mpmath>=0.19
  Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Collecting sniffio>=1.1
  Downloading sniffio-1.3.0-py3-none-any.whl (10 kB)
Collecting exceptiongroup
  Downloading exceptiongroup-1.1.2-py3-none-any.whl (14 kB)
Requirement already satisfied: mypy-extensions>=0.3.0 in /databricks/python3/lib/python3.10/site-packages (from typing-inspect->pyre-extensions==0.0.29->xformers>=0.0.19->vllm) (0.4.3)
Building wheels for collected packages: vllm
  Building wheel for vllm (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for vllm (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [179 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/utils.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/config.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/block.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/logger.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/outputs.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/sequence.py -> build/lib.linux-x86_64-cpython-310/vllm
      copying vllm/sampling_params.py -> build/lib.linux-x86_64-cpython-310/vllm
      creating build/lib.linux-x86_64-cpython-310/vllm/model_executor
      copying vllm/model_executor/model_loader.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor
      copying vllm/model_executor/utils.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor
      copying vllm/model_executor/weight_utils.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor
      copying vllm/model_executor/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor
      copying vllm/model_executor/input_metadata.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor
      creating build/lib.linux-x86_64-cpython-310/vllm/core
      copying vllm/core/block_manager.py -> build/lib.linux-x86_64-cpython-310/vllm/core
      copying vllm/core/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/core
      copying vllm/core/scheduler.py -> build/lib.linux-x86_64-cpython-310/vllm/core
      copying vllm/core/policy.py -> build/lib.linux-x86_64-cpython-310/vllm/core
      creating build/lib.linux-x86_64-cpython-310/vllm/engine
      copying vllm/engine/ray_utils.py -> build/lib.linux-x86_64-cpython-310/vllm/engine
      copying vllm/engine/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/engine
      copying vllm/engine/arg_utils.py -> build/lib.linux-x86_64-cpython-310/vllm/engine
      copying vllm/engine/async_llm_engine.py -> build/lib.linux-x86_64-cpython-310/vllm/engine
      copying vllm/engine/llm_engine.py -> build/lib.linux-x86_64-cpython-310/vllm/engine
      creating build/lib.linux-x86_64-cpython-310/vllm/entrypoints
      copying vllm/entrypoints/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/entrypoints
      copying vllm/entrypoints/llm.py -> build/lib.linux-x86_64-cpython-310/vllm/entrypoints
      copying vllm/entrypoints/api_server.py -> build/lib.linux-x86_64-cpython-310/vllm/entrypoints
      creating build/lib.linux-x86_64-cpython-310/vllm/worker
      copying vllm/worker/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/worker
      copying vllm/worker/worker.py -> build/lib.linux-x86_64-cpython-310/vllm/worker
      copying vllm/worker/cache_engine.py -> build/lib.linux-x86_64-cpython-310/vllm/worker
      creating build/lib.linux-x86_64-cpython-310/vllm/transformers_utils
      copying vllm/transformers_utils/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils
      copying vllm/transformers_utils/config.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils
      copying vllm/transformers_utils/tokenizer.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils
      creating build/lib.linux-x86_64-cpython-310/vllm/model_executor/layers
      copying vllm/model_executor/layers/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/layers
      copying vllm/model_executor/layers/sampler.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/layers
      copying vllm/model_executor/layers/layernorm.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/layers
      copying vllm/model_executor/layers/attention.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/layers
      copying vllm/model_executor/layers/activation.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/layers
      creating build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils
      copying vllm/model_executor/parallel_utils/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils
      copying vllm/model_executor/parallel_utils/parallel_state.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils
      creating build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/opt.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/mpt.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/bloom.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/gpt_j.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/falcon.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/baichuan.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/gpt2.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/gpt_bigcode.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/gpt_neox.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      copying vllm/model_executor/models/llama.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/models
      creating build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils/tensor_parallel
      copying vllm/model_executor/parallel_utils/tensor_parallel/random.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils/tensor_parallel
      copying vllm/model_executor/parallel_utils/tensor_parallel/utils.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils/tensor_parallel
      copying vllm/model_executor/parallel_utils/tensor_parallel/mappings.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils/tensor_parallel
      copying vllm/model_executor/parallel_utils/tensor_parallel/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils/tensor_parallel
      copying vllm/model_executor/parallel_utils/tensor_parallel/layers.py -> build/lib.linux-x86_64-cpython-310/vllm/model_executor/parallel_utils/tensor_parallel
      creating build/lib.linux-x86_64-cpython-310/vllm/entrypoints/openai
      copying vllm/entrypoints/openai/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/entrypoints/openai
      copying vllm/entrypoints/openai/protocol.py -> build/lib.linux-x86_64-cpython-310/vllm/entrypoints/openai
      copying vllm/entrypoints/openai/api_server.py -> build/lib.linux-x86_64-cpython-310/vllm/entrypoints/openai
      creating build/lib.linux-x86_64-cpython-310/vllm/transformers_utils/configs
      copying vllm/transformers_utils/configs/mpt.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils/configs
      copying vllm/transformers_utils/configs/falcon.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils/configs
      copying vllm/transformers_utils/configs/__init__.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils/configs
      copying vllm/transformers_utils/configs/baichuan.py -> build/lib.linux-x86_64-cpython-310/vllm/transformers_utils/configs
      running build_ext
      building 'vllm.cache_ops' extension
      creating /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310
      creating /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/csrc
      Emitting ninja build file /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/build.ninja...
      Compiling objects...
      Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
      [1/2] /usr/local/cuda/bin/nvcc  -I/databricks/python/lib/python3.10/site-packages/torch/include -I/databricks/python/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -I/databricks/python/lib/python3.10/site-packages/torch/include/TH -I/databricks/python/lib/python3.10/site-packages/torch/include/THC -I/usr/local/cuda/include -I/local_disk0/.ephemeral_nfs/envs/pythonEnv-f0090f6f-4b0c-4cce-a889-5dbaac3ea2f4/include -I/usr/include/python3.10 -c -c /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/csrc/cache_kernels.cu -o /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/csrc/cache_kernels.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O2 -std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -gencode arch=compute_75,code=sm_75 --threads 8 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=cache_ops -D_GLIBCXX_USE_CXX11_ABI=0
      FAILED: /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/csrc/cache_kernels.o
      /usr/local/cuda/bin/nvcc  -I/databricks/python/lib/python3.10/site-packages/torch/include -I/databricks/python/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -I/databricks/python/lib/python3.10/site-packages/torch/include/TH -I/databricks/python/lib/python3.10/site-packages/torch/include/THC -I/usr/local/cuda/include -I/local_disk0/.ephemeral_nfs/envs/pythonEnv-f0090f6f-4b0c-4cce-a889-5dbaac3ea2f4/include -I/usr/include/python3.10 -c -c /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/csrc/cache_kernels.cu -o /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/csrc/cache_kernels.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O2 -std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -gencode arch=compute_75,code=sm_75 --threads 8 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=cache_ops -D_GLIBCXX_USE_CXX11_ABI=0
      In file included from /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/csrc/cache_kernels.cu:2:
      /databricks/python/lib/python3.10/site-packages/torch/include/ATen/cuda/CUDAContext.h:6:10: fatal error: cusparse.h: No such file or directory
          6 | #include <cusparse.h>
            |          ^~~~~~~~~~~~
      compilation terminated.
      In file included from /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/csrc/cache_kernels.cu:2:
      /databricks/python/lib/python3.10/site-packages/torch/include/ATen/cuda/CUDAContext.h:6:10: fatal error: cusparse.h: No such file or directory
          6 | #include <cusparse.h>
            |          ^~~~~~~~~~~~
      compilation terminated.
      [2/2] c++ -MMD -MF /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/csrc/cache.o.d -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/databricks/python/lib/python3.10/site-packages/torch/include -I/databricks/python/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -I/databricks/python/lib/python3.10/site-packages/torch/include/TH -I/databricks/python/lib/python3.10/site-packages/torch/include/THC -I/usr/local/cuda/include -I/local_disk0/.ephemeral_nfs/envs/pythonEnv-f0090f6f-4b0c-4cce-a889-5dbaac3ea2f4/include -I/usr/include/python3.10 -c -c /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/csrc/cache.cpp -o /tmp/pip-install-6r7z9gxl/vllm_6909bb0353694b988c8efe2c971a3645/build/temp.linux-x86_64-cpython-310/csrc/cache.o -g -O2 -std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=cache_ops -D_GLIBCXX_USE_CXX11_ABI=0
      ninja: build stopped: subcommand failed.
      Traceback (most recent call last):
        File "/databricks/python/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 1900, in _run_ninja_build
          subprocess.run(
        File "/usr/lib/python3.10/subprocess.py", line 524, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.
      
      The above exception was the direct cause of the following exception:
      
      Traceback (most recent call last):
        File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-f0090f6f-4b0c-4cce-a889-5dbaac3ea2f4/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-f0090f6f-4b0c-4cce-a889-5dbaac3ea2f4/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-f0090f6f-4b0c-4cce-a889-5dbaac3ea2f4/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 261, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/databricks/python/lib/python3.10/site-packages/setuptools/build_meta.py", line 244, in build_wheel
          return self._build_with_temp_dir(['bdist_wheel'], '.whl',
        File "/databricks/python/lib/python3.10/site-packages/setuptools/build_meta.py", line 229, in _build_with_temp_dir
          self.run_setup()
        File "/databricks/python/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
          exec(code, locals())
        File "<string>", line 145, in <module>
        File "/databricks/python/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 973, in run_commands
          self.run_command(cmd)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
          cmd_obj.run()
        File "/databricks/python/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 299, in run
          self.run_command('build')
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
          self.distribution.run_command(command)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
          cmd_obj.run()
        File "/databricks/python/lib/python3.10/site-packages/setuptools/command/build.py", line 24, in run
          super().run()
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 132, in run
          self.run_command(cmd_name)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
          self.distribution.run_command(command)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
          cmd_obj.run()
        File "/databricks/python/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 79, in run
          _build_ext.run(self)
        File "/databricks/python/lib/python3.10/site-packages/Cython/Distutils/old_build_ext.py", line 186, in run
          _build_ext.build_ext.run(self)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 346, in run
          self.build_extensions()
        File "/databricks/python/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 843, in build_extensions
          build_ext.build_extensions(self)
        File "/databricks/python/lib/python3.10/site-packages/Cython/Distutils/old_build_ext.py", line 195, in build_extensions
          _build_ext.build_ext.build_extensions(self)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 466, in build_extensions
          self._build_extensions_serial()
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 492, in _build_extensions_serial
          self.build_extension(ext)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 202, in build_extension
          _build_ext.build_extension(self, ext)
        File "/databricks/python/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 547, in build_extension
          objects = self.compiler.compile(
        File "/databricks/python/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 658, in unix_wrap_ninja_compile
          _write_ninja_file_and_compile_objects(
        File "/databricks/python/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 1573, in _write_ninja_file_and_compile_objects
          _run_ninja_build(
        File "/databricks/python/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 1916, in _run_ninja_build
          raise RuntimeError(message) from e
      RuntimeError: Error compiling objects for extension
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for vllm
Failed to build vllm
ERROR: Could not build wheels for vllm, which is required to install pyproject.toml-based projects

[notice] A new release of pip available: 22.2.2 -> 23.2.1
[notice] To update, run: pip install --upgrade pip
```

**Comment by alexradu90** [comment]
+1 facing the same error in Databricks with DBR 13.1

**Comment by WoosukKwon** [comment]
Hi @patrafter1999 @alexradu90, thanks for trying out vLLM. I believe this should now be fixed as vLLM is shipped with pre-compiled CUDA binaries since v0.1.4.
